x-pack/plugins/security_solution/public/timelines/containers/helpers.ts

```diff
@@ -4,14 +4,32 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
+import { isEqual } from 'lodash/fp';
 import { TimelineId } from '../../../common/types/timeline';
 
 export const detectionsTimelineIds = [
   TimelineId.detectionsPage,
   TimelineId.detectionsRulesDetailsPage,
 ];
 
-export const skipQueryForDetectionsPage = (id: string, defaultIndex: string[]) =>
-  id != null &&
+/**
+ * If the defaultIndex for timeline and the defaultKibana index are equal to each other
+ * then this does not contain the signals index and we would want to exclude it completely from a query
+ * while on the detections page or the detection details page. This is to prevent additional querying
+ * from happening with the detections page and detections details when the default index is pushed into timeline.
+ * This code is considered a workaround and should go away in the future after refactoring/rewrites of timeline.
+ * @param id The id of the page which if it is set to the detections page or details page then it should not be analyzed
+ * @param defaultIndex The default index that is about cause a query from timeline.
+ * @param defaultKibanaIndex The default Kibana index to check to see if they're the same thing and if so, let us not query
+ */
+export const skipQueryForDetectionsPage = ({
+  id,
+  defaultIndex,
+  defaultKibanaIndex,
+}: {
+  id: string;
+  defaultIndex: string[];
+  defaultKibanaIndex: string[];
+}) =>
   detectionsTimelineIds.some((timelineId) => timelineId === id) &&
-  !defaultIndex.some((di) => di.toLowerCase().startsWith('.siem-signals'));
+  isEqual(defaultIndex, defaultKibanaIndex);
```

Comment:
From what I can tell `isEqual` will take order into account (which I don't believe matters here), so we'll want to `sort()` these before comparing, no?

Reply:
Yes, that is true. I didn't want to do reference equals or use .join(), but you are correct. I will add a test case for that specifically which fails and change this